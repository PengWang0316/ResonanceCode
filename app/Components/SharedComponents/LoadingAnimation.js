import React from 'react';
import { connect } from 'react-redux';

const LoadingAnimation = props => {
  if (props.isLoading) return (
    <div className="w-100 h-100 d-flex justify-content-center align-items-center" style={{ minHeight: '150px' }}>
      <div className="progress" style={{ width: '80%' }}>
        <div className="progress-bar progress-bar-striped progress-bar-animated" role="progressbar" aria-valuenow="75" aria-valuemin="0" aria-valuemax="100" style={{ width: '100%' }}>Loading...</div>
      </div>
    </div>);
  return <div />;
};

const mapStateToProps = state => ({ isLoading: state.isLoading });
export default connect(mapStateToProps, null)(LoadingAnimation);
