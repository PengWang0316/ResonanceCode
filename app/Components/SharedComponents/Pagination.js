import React, { Component } from 'react';
import PropTypes from 'prop-types';

// import { NUMBER_OF_READING_PER_PAGE } from '../config';

/** The pagination for the page.
  *Keeping amount and fetchContent out of Pagination component can help it raising reusability.
*/
class Pagination extends Component {
  static propTypes = {
    numberPerpage: PropTypes.number.isRequired,
    amount: PropTypes.number.isRequired,
    fetchContent: PropTypes.func.isRequired
  };

  state = { currentPage: 0 };

  /** Calculating the .
    * @return {null} No return.
  */
  componentWillMount() {
    // this.totalPage = this.props.amount / this.props.numberPerpage < 1 ?
    //   1 : this.props.amount / this.props.numberPerpage;
    this.currentPage = 0;
    this.initialPaginationArray(this.props.amount, this.props.numberPerpage);
    // this.assemblePaginationArray();
  }

  /** ReAssemble page number when the component receives a new totalPage number.
    * @param {object} nextProps is an object that contains new props' values.
    * @return {null} No return.
  */
  componentWillReceiveProps(nextProps) {
    if (this.props.amount !== nextProps.amount)
      this.initialPaginationArray(nextProps.amount, nextProps.numberPerpage);
  }

  /** Calculating the totalPage and call assembling method.
    * @param {number} amount is the total amount of the content will be showed.
    * @param {number} numberPerpage is the number amount will be showed in one page.
    * @return {null} No return.
  */
  initialPaginationArray(amount, numberPerpage) {
    this.totalPage = amount / numberPerpage < 1 ?
      1 : amount / numberPerpage;
    this.assemblePaginationArray();
  }

  /** Assembling an array for the pagination elements.
    * @return {null} No return.
  */
  assemblePaginationArray() {
    this.state.paginationArray = [];
    for (let i = 0; i < this.totalPage; i++) this.state.paginationArray.push(<li key={i} className={this.currentPage === i ? 'page-item active' : 'page-item'}><a href="#" role="button" tabIndex="-2" onClick={this.handleClickPageNumber} className="page-link">{i + 1}</a></li>);
    this.setState({ paginationArray: this.state.paginationArray });
  }

  /** Fetching the reading information when a user click the page number.
    * Also, set the currentPage state to this number.
    * @param {object} event is an object that comes from the element was clicked.
    * @return {null} No return.
  */
  handleClickPageNumber = ({ target }) => {
    if (this.currentPage !== (target.innerText * 1) - 1) {
      this.currentPage = (target.innerText * 1) - 1;
      this.props.fetchContent(this.currentPage);
      this.setState({ currentPage: this.currentPage });
      this.assemblePaginationArray();
    }
  };

  /** Going to preverious page when a user clicks the previous button.
    * @return {null} No return.
  */
  goPreviousPage = () => {
    this.currentPage = this.state.currentPage - 1;
    this.props.fetchContent(this.currentPage);
    this.setState({ currentPage: this.currentPage });
    this.assemblePaginationArray();
  };

  /** Going to next page when a user clicks the previous button.
    * @return {null} No return.
  */
  goNextPage = () => {
    this.currentPage = this.state.currentPage + 1;
    this.props.fetchContent(this.currentPage);
    this.setState({ currentPage: this.currentPage });
    this.assemblePaginationArray();
  };

  /** The render method for the component.
    * @return {jsx} Return the jsx for the component.
  */
  render() {
    return (
      <nav aria-label="Page navigation example">
        <ul className="pagination pagination-sm justify-content-end">
          <li className={this.state.currentPage === 0 ? 'page-item disabled' : 'page-item'}>
            <a href="#" role="button" onClick={this.goPreviousPage} className="page-link" tabIndex="-1">Previous</a>
          </li>
          {this.state.paginationArray}
          <li className={(this.state.currentPage + 1) * this.props.numberPerpage >= this.props.amount ? 'page-item disabled' : 'page-item'}>
            <a href="#" onClick={this.goNextPage} role="button" className="page-link">Next</a>
          </li>
        </ul>
      </nav>
    );
  }
}
export default Pagination;
