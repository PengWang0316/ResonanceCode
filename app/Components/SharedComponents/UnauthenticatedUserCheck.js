import React from 'react';
import { connect } from 'react-redux';

const UnauthenticatedUserCheck = props => (
  <div>
    {props.user.isLoaded && !props.user.isAuth &&
      <span>Unauthenticated action. Please login and try it again</span>}
    {props.user.isLoaded && props.user.isAuth && props.children}
  </div>
);
const mapStateToProps = state => ({ user: state.user });
export default connect(mapStateToProps, null)(UnauthenticatedUserCheck);
