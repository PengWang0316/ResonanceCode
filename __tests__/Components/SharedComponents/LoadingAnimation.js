import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import { LoadingAnimation } from '../../../app/Components/SharedComponents/LoadingAnimation';

describe('LoadingAnimation test', () => {
  const defaultProps = {
    isLoading: true
  };
  test('LoadingAnimation is loading snapshot', () => expect(renderer.create(<LoadingAnimation {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('LoadingAnimation is not loading snapshot', () => expect(renderer.create(<LoadingAnimation isLoading={false} />).toJSON()).toMatchSnapshot());
});
