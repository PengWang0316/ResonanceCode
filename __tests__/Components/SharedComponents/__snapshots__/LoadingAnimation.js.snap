// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation test LoadingAnimation isLoading snapshot 1`] = `
<div
  className="w-100 h-100 d-flex justify-content-center align-items-center"
  style={
    Object {
      "minHeight": "150px",
    }
  }
>
  <div
    className="progress"
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="75"
      className="progress-bar progress-bar-striped progress-bar-animated"
      role="progressbar"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`LoadingAnimation test LoadingAnimation not isLoading snapshot 1`] = `<div />`;
